saleor/graphql/core/types/taxes.py

```diff
@@ -384,15 +398,27 @@ def resolve_discounts(root: Union[Checkout, Order], info: ResolveInfo):
             def calculate_checkout_discounts(checkout_info):
                 checkout = checkout_info.checkout
                 discount_name = checkout.discount_name
-                return (
-                    [{"name": discount_name, "amount": checkout.discount}]
-                    if checkout.discount
-                    and (
-                        is_order_level_voucher(checkout_info.voucher)
-                        or has_checkout_order_promotion(checkout_info)
-                    )
-                    else []
-                )
+
+                if not checkout.discount:
+                    return []
+
+                has_order_promotion = has_checkout_order_promotion(checkout_info)
+                has_order_level_voucher = is_order_level_voucher(checkout_info.voucher)
+                # order promotion (currently we handle SUBTOTAL_PROMOTION only) applies
+                # to subtotal only
+                distribute_over_shipping = not has_order_promotion
+
+                if has_order_level_voucher or has_order_promotion:
+                    return [
+                        {
+                            "name": discount_name,
+                            "amount": checkout.discount,
+                            "distribute_over_subtotal": True,
```

Comment:
What is the reason for sending a flag which always is `True`? If you provide True/False flag it also means, that the app needs to have a logic to handle `False` action - app owner will not go through the core code to check that this flag is actually always set to True. Also app would need to cover different combination of these flags.
Also having these two flags means less extensibility. For example, what if you will have a third condition to cover new discount for `TaxableObjectDiscount` ? Then you will need to add third flag (or even more). 
Can't we create a new ENUM like `type` that would store the details of these two flags? You could go even further and provide there a details like: `MANUAL`, `VOUCHER`, `ORDER_PROMOTION` etc. Then each value could get a proper description, of what needs to be done.